src/TabBarIndicator.tsx

```diff
@@ -6,39 +6,107 @@ import memoize from './memoize';
 import { Route, SceneRendererProps, NavigationState } from './types';
 
 export type Props<T extends Route> = SceneRendererProps & {
+  dynamicWidth?: boolean;
```

Comment:
I'm not a fan of the indicator being aware of this prop. Maybe a `getTabWidth` prop which accepts an index and returns a width is cleaner. It could return the measured width or pre-calculated (layout.width / routes.length) depending on if the widths are dynamic